lib/indico/image.rb

```diff
@@ -0,0 +1,122 @@
+require 'oily_png'
+require 'base64'
+
+module Indico
+  def self.preprocess(image, size, batch)
+    # image is [[f,f,f,f, ...], ..] or [[[f,f,f],[f,f,f],[f,f,f]], ...]
+    if batch
+      # Batch Request
+      im_array = Array.new
+
+      # process each image
+      image.each do |_image|
+        im_array.push(preprocess(_image, size, false))
+      end
+
+      return im_array
+    end
+
+    if image.class == String
+      decoded_image = handle_string_input(image)
+    elsif image.class == Array
+      decoded_image = handle_array_input(image)
+    else
+      raise Exception.new("Image input must be nested array of pixels, filename, or base64 string")
+    end
+
+    # Resize and export base64 encoded string
+    return decoded_image.resize(size, size).to_data_url.gsub("data:image/png;base64," ,"")
+  end
+
+  def self.handle_string_input(str)
+    # Handles string input
+    if File.file?(str)
+      # Handling File Inputs
+      image = load_image(str)
+      return handle_array_input(image)
+    end
+
+    regex = %r{^([A-Za-z0-9+/]{4})*([A-Za-z0-9+/]{4}|[A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{2}==)$}.match(str)
+    # Base 64 handling
+    if regex == nil
+      raise Exception.new("String is not a valid absolute filepath or base64 string")
+    else
+      return ChunkyPNG::Image.from_data_url("data:image/png;base64," + str.gsub("data:image/png;base64," ,""))
+    end
+  end
+
+  def self.handle_array_input(image)
+    # Handles properly formatting and loading array of pixels
```

Comment:
Should we do the same thing as python here and let people know that array inputs won't be supported long-term?